### DISCUSSION TITLE
as_tree() function for Widget?

### DISCUSSION THREAD

**Discussion by peter-scholtens** [fltk-rs/fltk-rs#1263] [Ideas]
```
{
    let mut grp = fltk::group::Group::default();
    grp.begin();
    let mut tree = Tree::default_fill();
    grp.end();
}
...
let grp.child(0).unwrap().get_items();
```
If I attach a tree widget to a group, somewhere in a function call or other location with limited scope, how can I get access to the tree widget while scanning the children of the group? The function `child()` of the `Group` widget returns a  `Option<Widget>`. Earlier I solved this by using a `as_group()` type transformation. So would it be handy to have a `as_tree()` too?
In the code example above, the `get_items()` call fails, as the unwrap return a widget and **not** a tree.

**Comment by MoAlyousef** [comment]
The as_window(), as_group() methods are part of FLTK proper. Unfortunately there is no as_tree() method. So the children widgets are type-erased, and the only way to get back the type is by casting them back (unsafe, using into_widget()).
Otherwise you can keep a reference to the tree (outside of that scope or via statics), then when you add the tree, resize it to fill the group.

**Reply by peter-scholtens** [reply]
Clear! Indeed, these are the two options left I could think of.
